pkg/controller/job/jobcontroller.go

```diff
@@ -327,15 +332,15 @@ func (jm *JobController) syncJob(key string) error {
 	// the store after we've checked the expectation, the job sync is just deferred till the next relist.
 	jobKey, err := controller.KeyFunc(&job)
 	if err != nil {
-		glog.Errorf("Couldn't get key for job %#v: %v", job, err)
-		return err
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for job %#v: %v", job, err))
+		// We explicitly return nil to avoid re-enqueue the bad key
+		return nil
 	}
 	jobNeedsSync := jm.expectations.SatisfiedExpectations(jobKey)
 	selector, _ := unversioned.LabelSelectorAsSelector(job.Spec.Selector)
 	pods, err := jm.podStore.Pods(job.Namespace).List(selector)
 	if err != nil {
-		glog.Errorf("Error getting pods for job %q: %v", key, err)
-		jm.queue.Add(key)
+		utilruntime.HandleError(fmt.Errorf("Error getting pods for job %q: %v", key, err))
```

Comment:
Don't print the error twice.  Just return the error from here and let the higher level logic report it.